pkg/util/condition.go

```diff
@@ -19,21 +19,45 @@ package util
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
 )
 
-// TryAppendCondition appends the given Condition if it's not equal to the last Condition. Returns a new array if appended,
-// otherwise nil.
-func TryAppendCondition(conditions []metav1.Condition, newCondition metav1.Condition) []metav1.Condition {
+// TryAppendCondition appends the given Condition if it's not equal to the last Condition.
+func TryAppendCondition(conditions *[]metav1.Condition, newCondition metav1.Condition) {
```

Comment:
Perhaps `AppendConditionIfChanged` would give a better idea of what the function does. Can’t we keep the existing function semantics, which match `append`’s? _I.e._ return the slice...

Reply:
Even your suggestion doesn't really convey the exact meaning (ie if what changed?). What we really want to say is `AppendConditionIfNotEqualToTheLast` but that's long. I thought `Try` reasonably conveyed that enough w/o being too long or becoming a sentence.

Originally I thought the caller would want to know if it was actually appended but that didn't turn out to be the case. I agree it makes sense to return the slice.